src/components/PressableWithDelayToggle.js

```diff
@@ -0,0 +1,148 @@
+    styles: PropTypes.arrayOf(PropTypes.object),
+
+    /** Styles to apply to the text */
+    // eslint-disable-next-line react/forbid-prop-types
+    textStyles: PropTypes.arrayOf(PropTypes.object),
+
+    /** Styles to apply to the icon */
+    // eslint-disable-next-line react/forbid-prop-types
+    iconStyles: PropTypes.arrayOf(PropTypes.object),
+
+    /** Callback to be called on onPress */
+    onPress: PropTypes.func.isRequired,
+
+    /** The icon to display */
+    icon: PropTypes.func,
+
+    /** The icon to display once the pressable is pressed */
+    iconChecked: PropTypes.func,
+
+    /** If the component should be inline with text or not */
+    inline: PropTypes.bool,
+
+    ...withLocalizePropTypes,
+
+    ...withDelayToggleButtonStatePropTypes,
+};
+
+const defaultProps = {
+    text: '',
+    textChecked: '',
+    tooltipText: '',
+    tooltipTextChecked: '',
+    styles: [],
+    textStyles: [],
+    iconStyles: [],
+    icon: null,
+    inline: true,
+    iconChecked: Expensicons.Checkmark,
+};
+
+function PressableWithDelayToggle(props) {
+    const updatePressState = () => {
+        if (props.isDelayButtonStateComplete) {
+            return;
+        }
+        props.toggleDelayButtonState(true);
+        props.onPress();
+    };
+
+    const children = (
+        <Tooltip
+            containerStyles={styles.flexRow}
+            text={props.isDelayButtonStateComplete ? props.tooltipTextChecked : props.tooltipText}
+        >
+            <Text
+                suppressHighlighting
+                style={[...props.textStyles, styles.mr1]}
+            >
+                {props.isDelayButtonStateComplete && props.textChecked ? props.textChecked : props.text}
+            </Text>
+            <Pressable
+                ref={props.innerRef}
+                focusable
+                accessibilityLabel={props.isDelayButtonStateComplete ? props.tooltipTextChecked : props.tooltipText}
+                onPress={updatePressState}
+            >
+                {({hovered, pressed}) => (
+                    <>
+                        {props.icon && (
+                            <Icon
+                                src={props.isDelayButtonStateComplete ? props.iconChecked : props.icon}
+                                fill={StyleUtils.getIconFillColor(getButtonState(hovered, pressed, props.isDelayButtonStateComplete))}
+                                style={props.iconStyles}
+                                width={variables.iconSizeSmall}
+                                height={variables.iconSizeSmall}
+                            />
+                        )}
+                    </>
+                )}
+            </Pressable>
+        </Tooltip>
+    );
+
+    // Due to limitations in RN regarding the vertical text alignment of non-Text elements,
+    // for elements that are supposed to be inline, we need to use a Text element instead
+    // of a Pressable
+    return props.inline ? (
```

Comment:
wouldn't work to store the component into a variable and use it?

```
const PressableView = props.inline ? Text : Pressable;

return (<PressableView 
                style={[...props.styles, styles.flexRow]}
                onPress={updatePressState}>
                      <Tooltip> [...] </Tooltip>
              </PressableView>);
```
I also would remove the `children` const because won't be needed

Reply:
LGTM just an small suggestion

Reply:
I agree, thank you!